globalconf.go

```diff
@@ -143,7 +143,7 @@ func (g *GlobalConf) Parse() {
 					case flag.ContinueOnError:
 						return
 					case flag.ExitOnError:
-						fmt.Printf("failed to set %s\n%s\n", f.Name, err)
+						fmt.Printf("failed to set flag, section: [%s], flag/setting: %s\nerror: %s\n", set.Name(), f.Name, err)
```

Comment:
this seems like an awkwardly verbose phrasing.
why not just say `fmt.Printf("failed to set %s.%s: %s", set.Name(), f.Name, err)`

Reply:
is it possible for the section to be empty? in that case we should print a different error that doesn't mention the section.

Reply:
For settings hat do not belong to a `section`/`FlagSet` it uses the `path` to the binary which comes from the golang `flag.go` source (`os.Args[0]`), which is the default for `CommandLine` flags. An example is as follows (using your suggested modification for the error message):
```
metrictank_1_20de441b0c86 | failed to set /usr/bin/metrictank.drop-first-chunk: strconv.ParseBool: parsing "invalidValue": invalid syntax
```
All non-default `section`/`FlagSet` will use the name of the `section`/`FlagSet` as it is when `Register` is called.

Reply:
hmm @woodsaj thought global vars would go under section `"default"`.
regardless whether it is default or the path, for global vars I think we shouldn't mention the section at all. is that something we can do?

Reply:
I think that would work unless someone uses `[default]` as one of their sections.

Reply:
This looks easy to fix to me. Instead of looking at `set.Name()`, just look at `name` which is already set to the section name or "" for settings not under a section.

So you could easily change it to
```
						var section string
						if name != "" {
							section = name + "."
						}
						fmt.Printf("failed to set %s%s: %s\n", section, f.Name, err)
```